src/messaging/ExchangeMgr.cpp

```diff
@@ -296,7 +296,7 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
         ChipLogProgress(ExchangeManager, "ec pos: %d, id: %d, Delegate: 0x%x", ec - mContextPool.begin(), ec->GetExchangeId(),
                         ec->GetDelegate());
 
-        ec->HandleMessage(packetHeader, payloadHeader, std::move(msgBuf));
+        ec->HandleMessage(packetHeader, payloadHeader, source, std::move(msgBuf));
```

Comment:
Same, ExchangeContext should only handle Node address instead of low level IP address

Reply:
Resolving it based on our slack conversation.
The address is needed as PASE session establishment using SoftAP provisioning needs to provide peer address.
SecureSessionHandle class could be refactored to carry this information. That will be done in a separate PR, as the scope of the refactor is greater than this PR.